modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java

```diff
@@ -1474,9 +1397,15 @@ public KBArticle updateStatus(
 
 		// Subscriptions
 
-		_notify(
-			Collections.singleton(_NOTIFICATION_RECEIVER_SUBSCRIBER), userId,
-			kbArticle, action, serviceContext);
+		Set<String> receivers = SetUtil.fromArray(
+			_NOTIFICATION_RECEIVER_SUBSCRIBER);
+
+		if (status == WorkflowConstants.STATUS_EXPIRED) {
+			action = Constants.EXPIRE;
+			receivers.add(_NOTIFICATION_RECEIVER_OWNER);
```

Comment:
I think you should add an extra newline after updating `action`.

Reply:
Also, the assignments to `action` seem to be all over the place. Can you centralize that computation in one method and remove all assignments to it? It's getting quite messy. E.g.:
```
private String _getAction(KBArticle kbArticle, int status) {
    if (status == WorkflowConstants.STATUS_EXPIRED) {
        return Constants.EXPIRE;
    }
    else if (kbArticle.isFirstVersion()) {
        return Constants.ADD;
    }
    else {
        return Constants.UPDATE;
    }
}
```

Reply:
I like this centralization. Can we remove `else` statements?
```java
private String _getAction(KBArticle kbArticle, int status) {
    if (status == WorkflowConstants.STATUS_EXPIRED) {
        return Constants.EXPIRE;
    }
    
    if (kbArticle.isFirstVersion()) {
        return Constants.ADD;
    }
    
    return Constants.UPDATE;
}
```